CONTRIBUTING.md

```diff
@@ -9,11 +9,146 @@ We'd love to accept your patches and contributions to this project.
 
-## Contribution process
-
-### Code reviews
+### Reviews
 
 All submissions, including submissions by project members, require review. We
 use GitHub pull requests for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using pull requests.
+
+### Setup
+
+Required steps:
+
+1. Create [a GitHub account](https://github.com/join).
+1. [Install git](https://help.github.com/articles/set-up-git/) for source control.
+1. Clone the repository, e.g:
+
+  ```shell
+  git clone git@github.com:ossf/malicious-packages.git
+  cd malicious-packages
+  ```
+
+Contributing changes will require forking the repository and pushing
+your changes to the fork.
+
+Optional:
+
+- [Install Go](https://go.dev/dl/) for code development.
+- Install `make` for running development commands.
+
+
+## Contributing OSV Reports
+
+We accept new reports of malicious packages that are within [scope](README.md#scope),
+and updates to any existing report in the database.
+
+**Note:** This process is new and untested, it will change and evolve
+over time. We welcome [suggestions](https://github.com/ossf/malicious-packages/issues/new)
+on how the process can be improved.
+
+### Via Pull Request
+
+#### New Individual Report
+
+1. If needed, create the directory under `./malicious/` for the report
+   to live in.
+    - Directories are formatted `[ecosystem]/[package_name]`, with no
+      escaping of slashes.
+      (e.g. `./malicious/go/github.com/ossf/package-analysis/` or
+      `./malicious/npm/@example/package/`).
+    - If you're unsure, once the PR is created this will be checked by a
+      GitHub action.
+1. Create the JSON file with the name `MAL-0000-[name].json`. Where
+   `[name]` can be replaced with any arbitrary text.
+1. Populate the JSON file conforming to the [OSV Schema](https://ossf.github.io/osv-schema/)
+   format and the [notes below](#osv-schema-notes).
+1. Commit and push the change to a fork and submit a Pull Request.
+    - Please fix any validation errors.
+
+#### Bulk Import
+
+1. [Create an issue](https://github.com/ossf/malicious-packages/issues/new),
+   which should include the following:
+    - A proposed unique "source id" if you do not already have one.
+    - Some details on the reports (i.e. quantity, ecosystems), etc.
+1. Use the `ingest` command to copy the reports from a local folder to the
+   appropriate locations, and to ensure the origin data is correctly populated.
+    - Ensure your OSV reports are in a local directory.
+    - From the repo run (requires [Go](https://go.dev/doc/install)):
+
+    ```shell
+    go run ./cmd/ingest \
+       -config config/config.yaml \
+       -dir /path/to/osv \
+       -source [source id]
+    ```
+
+    - Fix any issues. You will need to revert the repo before running
+      ingest again.
+1. Commit and push the changes to a form and submit a Pull Request.
+    - Reference the issue from step #1 in the PR.
```

Comment:
What if someone wants to do periodic bulk imports? Would they reference the same issue each time?  And that issue would stay open?

Reply:
I've tried to clarify this. Let me know what you think.